### PR TITLE
Add ImageMagick 7 checks to rmutil.c

### PR DESCRIPTION
**Pull request by dlemstra**

### CODE REVIEW

**Comment by mockdeep** [comment]
@dlemstra this test failure happened on the other branch, too. Any idea what is happening?

```
===============================================================================
Error: test_builtin(KernelInfoUT)
  RuntimeError: failed to acquire builtin kernel
/home/travis/build/rmagick/rmagick/test/KernelInfo.rb:57:in `builtin'
/home/travis/build/rmagick/rmagick/test/KernelInfo.rb:57:in `test_builtin'
     54:     assert_instance_of(Magick::KernelInfo, Magick::KernelInfo.builtin(Magick::LoGKernel, 'LoG:10,5'))
     55:     assert_instance_of(Magick::KernelInfo, Magick::KernelInfo.builtin(Magick::DoGKernel, 'DoG:10,5'))
     56:     assert_instance_of(Magick::KernelInfo, Magick::KernelInfo.builtin(Magick::BlurKernel, 'Blur:10,5,1'))
  => 57:     assert_instance_of(Magick::KernelInfo, Magick::KernelInfo.builtin(Magick::CometKernel, 'Comet:10,5,1'))
     58:   end
     59: end
===============================================================================
```

File: ext/RMagick/rmutil.c [hunk 1]
```diff
@@ -1063,7 +1117,15 @@ void rm_sync_image_options(Image *image, Info *info)
 
     if (info->colorspace != UndefinedColorspace)
     {
+#if defined(IMAGEMAGICK_7)
+        exception = AcquireExceptionInfo();
+        SetImageColorspace(image, info->colorspace, exception);
+        // We should not throw an exception in this method and that is
```
Comment (mockdeep) [inline ext/RMagick/rmutil.c]: Why do we not want to throw an exception here?

Comment (dlemstra) [reply ext/RMagick/rmutil.c]: Just pushed a patch to make it more clear. Does it answer your question?

Comment (mockdeep) [reply ext/RMagick/rmutil.c]: I see. Yep, makes sense!

Comment (mockdeep) [reply ext/RMagick/rmutil.c]: Not for this PR, but should there be a way to transmit the exception back up? Like maybe the caller should pass in an exception reference or something?

Comment (dlemstra) [reply ext/RMagick/rmutil.c]: We could probably fix this without breaking the public PR. But that should be something for a new PR.

**Comment by dlemstra** [comment]
I have no idea why `assert_instance_of(Magick::KernelInfo, Magick::KernelInfo.builtin(Magick::CometKernel, 'Comet:10,5,1'))` fails but it looks like some kind of race condition. It only seems to happen on that ancient version of ImageMagick. And for some reason it passed in the build now.

File: ext/RMagick/rmutil.c [hunk 2]
```diff
@@ -1268,7 +1367,16 @@ rm_exif_by_entry(Image *image)
             }
             memcpy(str+len, property+5, property_l-5);
             len += property_l - 5;
-            value = GetImageProperty(image,property);
+#if defined(IMAGEMAGICK_7)
+            value = GetImageProperty(image, property, exception);
+            if (rm_should_raise_exception(exception, RetainExceptionRetention))
+            {
+                xfree(str);
+                rm_raise_exception(exception);
```
Comment (mockdeep) [inline ext/RMagick/rmutil.c]: Something I've been wondering about is what happens when the exception is raised. Does this break out of the C code right here? Or does it continue running the function below and then Ruby does some magic to raise the exception in Ruby land?

Comment (dlemstra) [reply ext/RMagick/rmutil.c]: The C code inside `rm_magick_error` calls the raise method of `Class_ImageMagickError`. That raises the exception inside Ruby land and stops the execution of the method.

**Pull request merged**